### DISCUSSION TITLE
SHTC3 or aht10 +bmp280

### DISCUSSION THREAD

**Discussion by jokokiswan** [openLuat/LuatOS#78] [Ideas]
I know luat support reading sht30 sensor, it got built in 0x44 addr on sht30 reading (shtc 0x70), does in near future it got shtc3 or aht10 + bmp?. thank you.
![151525kgq93cl3kyk1se9x](https://user-images.githubusercontent.com/58895576/209891294-e0e9eb2f-f353-4cb5-9ff7-719b446c4aa1.png)

**Comment by jokokiswan** [comment]
i2c_speed = i2c.SLOW
            i2c.setup(i2cid, i2c_speed)
            i2c.writeReg(0, 0x70, 0xE0, string.char(0x35, 0x17, 0x5C, 0x24))
            sys.wait(200)
            local data = i2c.readReg(0, 0x70, 0xE1, 6)
            sys.wait(100)
            i2c.writeReg(0, 0x70, 0Xb0, 0X98)

i tried walk arround with luat i2c, always receiving -1 data :), does it not right procedure?

**Comment by chenxuuu** [comment]
可以用 逻辑分析仪 或者 示波器 检查一下有没有实际的波形、实际的波形是否符合预期



---Original---
From: ***@***.***&gt;
Date: Thu, Dec 29, 2022 09:25 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: Re: [openLuat/LuatOS] SHTC3 or aht10 +bmp280 (Discussion #78)




         i2c_speed = i2c.SLOW         i2c.setup(i2cid, i2c_speed)         i2c.writeReg(0, 0x70, 0xE0, string.char(0x35, 0x17, 0x5C, 0x24))         sys.wait(200)         local data = i2c.readReg(0, 0x70, 0xE1, 6)         sys.wait(100)         i2c.writeReg(0, 0x70, 0Xb0, 0X98)  
i tried walk arround with luat i2c, always receiving -1 data :), does it not right procedure?

**Reply by jokokiswan** [reply]
OK

**Reply by jokokiswan** [reply]
I did not see if luatos already support aht10 sensor, thank you luatos team!

**Reply by chenxuuu** [reply]
https://wiki.luatos.com/api/libs/aht10.html

lua lib
